src/plugins/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -0,0 +1,392 @@
+  const extensionByType = extensions.reduce((acc, m) => {
+    return m!.docTypes.reduce((_acc, type) => {
+      acc[type] = m;
+      return acc;
+    }, acc);
+  }, {} as { [visType: string]: VisualizationsAppExtension });
+  const searchOption = (field: string, ...defaults: string[]) =>
+    _(extensions).map(field).concat(defaults).compact().flatten().uniq().value() as string[];
+  const searchOptions: SavedObjectsFindOptions = {
+    type: searchOption('docTypes', 'visualization'),
+    searchFields: searchOption('searchFields', 'title^3', 'description'),
+    search: search ? `${search}*` : undefined,
+    perPage: size,
+    page: 1,
+    defaultSearchOperator: 'AND' as 'AND',
+    hasReference: references,
+  };
+
+  const { total, savedObjects } = await savedObjectsClient.find<SavedObjectAttributes>(
+    searchOptions
+  );
+
+  return {
+    total,
+    hits: savedObjects.map((savedObject) => {
+      const config = extensionByType[savedObject.type];
+
+      if (config) {
+        return {
+          ...config.toListItem(savedObject),
+          references: savedObject.references,
+        };
+      } else {
+        return mapHitSource(visTypes, savedObject);
+      }
+    }),
+  };
+}
+
+export async function getSavedVisualization(
+  services: {
+    savedObjectsClient: SavedObjectsClientContract;
+    search: DataPublicPluginStart['search'];
+    dataViews: DataPublicPluginStart['dataViews'];
+    spaces: SpacesPluginStart;
+    savedObjectsTagging?: SavedObjectsTaggingApi;
+  },
+  opts?: any
+): Promise<VisSavedObject> {
+  if (typeof opts !== 'object') {
+    opts = { id: opts };
+  }
+
+  const id = (opts.id as string) || '';
+  const savedObject = {
+    id,
+    copyOnSave: false,
+    migrationVersion: opts.migrationVersion,
+    displayName: SAVED_VIS_TYPE,
+    getEsType: () => SAVED_VIS_TYPE,
+    getDisplayName: () => SAVED_VIS_TYPE,
+    searchSource: opts.searchSource ? services.search.searchSource.createEmpty() : undefined,
+  } as VisSavedObject & { _source: any };
+  const config = { injectReferences };
+
+  const defaultsProps = getDefaults(opts);
+
+  if (!id) {
+    Object.assign(savedObject, defaultsProps);
+    return savedObject;
+  }
+
+  const {
+    saved_object: resp,
+    outcome,
+    alias_target_id: aliasTargetId,
+  } = await services.savedObjectsClient.resolve<SavedObjectAttributes>(SAVED_VIS_TYPE, id);
+
+  if (!resp._version) {
+    throw new SavedObjectNotFound(SAVED_VIS_TYPE, id || '');
+  }
+
+  savedObject._source = _.cloneDeep(resp.attributes);
+
+  if (savedObject._source.visState) {
+    savedObject._source.visState = JSON.parse(savedObject._source.visState);
+  }
+
+  // assign the defaults to the response
+  _.defaults(savedObject._source, defaultsProps);
+
+  Object.assign(savedObject, savedObject._source);
+  savedObject.lastSavedTitle = savedObject.title;
+
+  savedObject.sharingSavedObjectProps = {
+    aliasTargetId,
+    outcome,
+    errorJSON:
```

Comment:
@VladLasitsa @flash1293  I know it was copied from `Lens`, but not sure that sharingSavedObjectProps should keep that `errorJSON` field. 

First of all `conflict` outcome is not an error and it's not semantically right to handle that case in `errorJSON`
What if we move it directly into place where we need it.  At least we can remove `spaces` dependency....

Reply:
Conflict is not an error, but in the context of not being able to display a vis in the dashboard because of conflict, I'd say it is an error from user perspective. Apart from that, that goes a further - there's a [maps PR](https://github.com/elastic/kibana/pull/112606/files) that unifies the usage of the savedObjectConflictMessage across lens and Maps uses the same semantics. 
When Maps PR is merged, it would be good to align this PR.  I didn't test the behavior yet, but from reading the code it looks like Visualize would not display the same message as Lens and maps (I might be wrong though):

<img width="256" alt="Screenshot 2021-10-04 at 11 04 05" src="https://user-images.githubusercontent.com/4283304/135823819-59349a08-23fe-4954-8d28-f5a004c60bbd.png">

Reply:
Now I provide  the following error message for embeddable: 
![image](https://user-images.githubusercontent.com/16915480/135831512-b0a84d1a-b22a-402a-8697-52c2ed2bd7fa.png)

Is it okay for now?